composer.json

```diff
@@ -26,7 +26,7 @@
         }
     ],
     "require": {
-        "php": "^7.2.5",
+        "php": "~7.2.22 || ^7.3.9",
```

Comment:
This syntax is way wrong. Please read https://developer.happyr.com/always-use-caret-instead-of-tilde

Reply:
IMO, the [Tilde Version Range](https://getcomposer.org/doc/articles/versions.md#tilde-version-range-) operator is the right choice for the desired version resolution. The constraint will respect SemVer, since it only allows to update to any minor or patch version from 7.x branch.
If you have any doubt about this behavior, please check https://semver.mwl.be/ ("php" is not recognized as a package there since it only handles the packages published in Packagist, but you can test with any other package and the result should be the same).

Reply:
Im sorry, you are correct.